kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -830,7 +830,12 @@ protected void handleProduceRequest(KafkaHeaderAndRequest produceHar,
         checkArgument(produceHar.getRequest() instanceof ProduceRequest);
         ProduceRequest produceRequest = (ProduceRequest) produceHar.getRequest();
 
-        final int numPartitions = produceRequest.data().topicData().size();
+        final int numPartitions = produceRequest
+                .data()
+                .topicData()
+                .stream()
```

Comment:
Streams can be less efficient than iterating directly and adding to a mutable variable. Given that this code is on the hot path, can we use a `for` loop?

Reply:
While a generally agree on this point, the new Java model for the wire protocol is using this pattern with multiple layers (topics > partitions).
Doing 2 'for' loops will require in any case a similar amount of resources, especially if the data structures are small. The gain probably is negligible.

Please note that we could create some utility method and refactor the codebase. I didn't do it because I wasn't sure that we would be axtuslly able to make the code more readable or efficient.

The code in this form is pretty compact and readable. 
I am leaning towards keeping it like this